### PR TITLE
Update agent tests for rhel5 support

### PR DESCRIPTION
**Pull request by barnabycourt**
ok test

### CODE REVIEW

**Comment by barnabycourt** [comment]
ok test

**Comment by jortel** [comment]
LGTM

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
ok test

**Comment by barnabycourt** [comment]
Closing this in favor of https://github.com/pulp/pulp/pull/1574 which is against the testing branch

File: agent/test/unit/test_agent.py [hunk 1]
```diff
@@ -837,7 +840,16 @@ def test_run_not_registered(self, validate, update_settings):
     @patch('pulp.agent.gofer.pulpplugin.update_settings')
     @patch('pulp.agent.gofer.pulpplugin.validate_registration')
     def test_run_validate_failed(self, validate, update_settings, sleep):
-        validate.side_effect = [ValueError, None]
+        # List is processed in reverse order
+        list_of_return_values = [None, True]
+
```
Comment (jortel) [inline agent/test/unit/test_agent.py]: I was thinking we may want to add support of this sort of thing in pulp.devel.unit.util.py
```
class SideEffect(object):
    """
    Side effect for older version of mock.
    """

    def __init__(self, *values):
        """
        Side effect values.

        Example:
          mock = Mock(side_effect=SideEffect(ValueError, None)

        :param values: List of values
        :type values: list
        """
        self.values = iter(values)

    def __call__(self):
        value = self.values.next()
        if callable(value):
            value = value()
        if isinstance(value, Exception):
            raise value
        return value
```

Then here do:

```
validate.side_effect = SideEffect(ValueError, None)

**Pull request closed without merging**